### PR TITLE
Fix inotify behavior when presented with non-ASCII path events

### LINKED ISSUE #516: Events from non-ASCII paths raise UncodeDecodeError
```
from os import mkdir
from watchdog.observers.inotify import InotifyObserver
from watchdog.events import FileSystemEventHandler

class H(FileSystemEventHandler):
    def on_any_event(self, event):
        print(event)

def main():
    d = u"demo"
    handler = H()

    mkdir(d)
    obs = InotifyObserver()
    obs.schedule(handler, d, recursive=True)
    obs.start()
    mkdir(u"/".join((d, u"\N{SNOWMAN}")))
    obs.join()

main()
```

results in this exception

```
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "watchdog/src/watchdog/observers/inotify_buffer.py", line 87, in run
    inotify_events = self._inotify.read_events()
  File "watchdog/src/watchdog/observers/inotify_c.py", line 324, in read_events
    inotify_event = InotifyEvent(wd, mask, cookie, name, src_path)
  File "watchdog/src/watchdog/observers/inotify_c.py", line 478, in __init__
    self._name = name.decode()
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 0: ordinal not in range(128)
```

Apparently introduced in 0c5e3531748d231e34ef6ecf8581ac20fd956b25

### PR DESCRIPTION
**Pull request by exarkun**
Fixes #516

### CODE REVIEW

**Comment by BoboTiG** [comment]
Hello @exarkun , thanks for the patch!

Could you change [that line](https://github.com/gorakhargosh/watchdog/blob/773d3da51a5ea1fcd8f0f33ca7445a4c24140f62/tox.ini#L13) to:
```python
python -bb -m pytest {posargs}
```
It will ensure there is no `BytesWarning`.

**Comment by exarkun** [comment]
> Hello @exarkun , thanks for the patch!
> 
> Could you change [that line](https://github.com/gorakhargosh/watchdog/blob/773d3da51a5ea1fcd8f0f33ca7445a4c24140f62/tox.ini#L13) to:
> 
> ```python
> python -bb -m pytest {posargs}
> ```
> 
> It will ensure there is no `BytesWarning`.

Sure thing.  Done.

File: tests/test_inotify_c.py [hunk 1]
```diff
@@ -115,3 +115,17 @@ def inotify_rm_watch(fd, wd):
...
+def test_non_ascii_path():
+    """
+    Inotify can construct an event for a path containing non-ASCII.
+    """
+    path = p(u"\N{SNOWMAN}")
+    with watching(p('')):
+        os.mkdir(path)
+        event, _ = event_queue.get(timeout=5)
+        assert isinstance(event.src_path, type(u""))
+        assert event.src_path == path
```
[the remark this answers is not in the document]
Comment (exarkun) [reply tests/test_inotify_c.py]: Please let me know when you've finished reviewing and I'll address all of your comments at once.

[the remark this answers is not in the document]
Comment (BoboTiG) [reply tests/test_inotify_c.py]: Done, sorry for the several steps :)

[the remark this answers is not in the document]
Comment (exarkun) [reply tests/test_inotify_c.py]: > assert event.src_path.endswith(u"\N{SNOWMAN}")  # or something similar to test the event path is the good one

I think this is amply covered by the line that's there already:

[the remark this answers is not in the document]
Comment (exarkun) [reply tests/test_inotify_c.py]: I've added the `assert repr(event)` though.

Thanks.

**Comment by BoboTiG** [comment]
Thanks for the changes, and I am sorry to ask you one last thing :)
I was trying to reproduce the error that was fixed with the commit you are partially reverting with this PR. This is simply to ensure no regression.

Could you apply those changes (I should have done it way sooner):
```diff
diff --git a/tests/conftest.py b/tests/conftest.py
index ed58af3..418a58d 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -20,3 +20,22 @@ def p(tmpdir, *args):
     with the provided arguments.
     """
     return partial(os.path.join, tmpdir)
+
+
+
+@pytest.fixture(autouse=True)
+def no_warnings(recwarn):
+    """Fail on warning."""
+
+    yield
+
+    warnings = []
+    for warning in recwarn:  # pragma: no cover
+        message = str(warning.message)
+        if (
+            "Not importing directory" in message
+            or "Using or importing the ABCs" in message
+        ):
+            continue
+        warnings.append("{w.filename}:{w.lineno} {w.message}".format(w=warning))
+    assert not warnings
```
With this fixture, you will see several failures related to your changes and this is what we tried to fix in the older commit. Many thanks for your patience and work :muscle:

**Comment by exarkun** [comment]
@BoboTiG The no-warnings fixture is now in place and I've fixed the test failures where such warnings were being triggered (probably; macOS travis jobs still running, the rest are green).

**Comment by BoboTiG** [comment]
Thanks @exarkun :)

**Pull request merged**